backend_new/src/services/multiselect-question.service.ts

```diff
@@ -0,0 +1,195 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaService } from './prisma.service';
+import { MultiselectQuestion } from '../dtos/multiselect-questions/multiselect-question-get.dto';
+import { MultiselectQuestionUpdate } from '../dtos/multiselect-questions/multiselect-question-update.dto';
+import { MultiselectQuestionCreate } from '../dtos/multiselect-questions/multiselect-question-create.dto';
+import { mapTo } from '../utilities/mapTo';
+import { SuccessDTO } from '../dtos/shared/success.dto';
+import { Prisma } from '@prisma/client';
+import { buildFilter } from '../utilities/build-filter';
+import { MultiselectQuestionFilterKeys } from '../enums/multiselect-questions/filter-key-enum';
+import { MultiselectQuestionQueryParams } from '../dtos/multiselect-questions/multiselect-question-query-params.dto';
+
+const view: Prisma.MultiselectQuestionsInclude = {
+  jurisdictions: true,
+};
+
+/*
+  this is the service for multiselect questions
+  it handles all the backend's business logic for reading/writing/deleting multiselect questione data
+*/
+@Injectable()
+export class MultiselectQuestionService {
+  constructor(private prisma: PrismaService) {}
+
+  /*
+    this will get a set of multiselect questions given the params passed in
+  */
+  async list(
+    params: MultiselectQuestionQueryParams,
+  ): Promise<MultiselectQuestion[]> {
+    const rawMultiselectQuestions =
+      await this.prisma.multiselectQuestions.findMany({
+        include: view,
+        where: this.buildWhere(params),
+      });
+    return mapTo(MultiselectQuestion, rawMultiselectQuestions);
+  }
+
+  /*
+    this will build the where clause for list()
+  */
+  buildWhere(
+    params: MultiselectQuestionQueryParams,
+  ): Prisma.MultiselectQuestionsWhereInput {
+    const filters: Prisma.MultiselectQuestionsWhereInput[] = [];
+    if (params?.filter?.length) {
```

Comment:
suggestion: remove this condition wrapping the logic and return sooner if params filter is not defined or has no length

Reply:
nice catch! updated!